### ISSUE TITLE
Multiple checkSession calls made by UserProvider when I want to skip the check completely

### ISSUE THREAD

**Issue by mderrick** [auth0/nextjs-auth0#754]
### Description

When SSR'ing there seem to be multiple calls made by the `UserProvider` to `/api/auth/me` even though the server has already determined there is no session - there is no way to tell it to skip.

### Reproduction

When rendering a route server side I can hydrate the user and pass it in as an initial user to the `UserProvider` which stops it making any requests to `/api/auth/me`:
```
// _app.ts
function MyApp({ Component, pageProps }: AppProps) {
      <UserProvider user={pageProps.user}>
        <Component {...pageProps} />
      </UserProvider>
    </>
  );
}
```

```
// pages/index.tsx
export const getServerSidePropsUser: GetServerSideProps =
  getServerSidePropsWrapper(async ({ req, res }) => {
    const session = getSession(req, res);
    return {
      props: {
        user: session?.user || null,
      },
    };
  });
```

However when there is no session, the client will make a request (twice for some reason) to `/api/auth/me` when it needn't make any at all as the server has already done the work. How can I skip the api call made by the `UserProvider` when I know it's going to 401.

### Environment

 "@auth0/nextjs-auth0": "^1.9.1"
 "next": "12.1.6"

**Comment by adamjmcgrath** [comment]
Hi @mderrick - thanks for raising this

> How can I skip the api call made by the UserProvider when I know it's going to 401.

Unfortunately, it is not current possible for the browser to know that the server has done the work if the user is not logged in. 

If `pageProps.user` is `undefined` in `<UserProvider user={pageProps.user}>` it may be because you've called `getSession` and `session?.user` is `undefined` and returned `undefined` as a prop from `getServerSideProps`. Or it may be that `pageProps.user` is `undefined` because you haven't checked the session at all for that specific page's `getServerSideProps`, then it's up to the browser to check. This is why the SDK will check once for a user on page load if none is provided.

If you don't want the browser to check the server and only base the context on what's been provided by the server from `getServerSideProps`, you don't really need the `UserProvider`, because this is pretty much [all it does](https://github.com/auth0/nextjs-auth0/blob/main/src/frontend/use-user.tsx#L187-L213). You could just use [useContext](https://reactjs.org/docs/hooks-reference.html#usecontext) directly.

>  the client will make a request (twice for some reason) 

The browser client shouldn't make a request twice, only once per page load. I've verified this on the example app, if you can create a reproducible example I'd be happy to take a look.

**Comment by adamjmcgrath** [comment]
Closing, feel free to ping me if you want to reopen this

**Comment by digkumar** [comment]
@adamjmcgrath 
When running the auth0-nextjs sample app, it makes checkSession (/api/auth/me) call twice. Is this expected. 
https://github.com/auth0-samples/auth0-nextjs-samples/tree/main/Sample-01
![image](https://github.com/auth0/nextjs-auth0/assets/153373674/dc533e8a-f3c4-4d15-ad9b-bc03687d9b43)

**Comment by hey-kevin** [comment]
@adamjmcgrath can you reopen this issue? I am seeing the `/api/auth/me` calls twice as well
